prettier: add page (#6192)
tldr-pages/tldr: pages/common/prettier.md

# prettier

> > An opinionated code formatter for JS, JSON, CSS, YAML and much more.

Reviewer: ```suggestion
> An opinionated code formatter for JS, JSON, CSS, YAML, and many more.
```

> > An opinionated code formatter for JS, JSON, CSS, YAML, and many more.

Reviewer 2: ```suggestion
> An opinionated code formatter for JavaScript, JSON, CSS, YAML, and more.
```

> More information: <https://prettier.io/>.

> - Format a file and print it to the stdout:

Reviewer 2: ```suggestion
- Format a file and print the result to stdout:
```

> `prettier {{file}}`

Reviewer 2: ```suggestion
`prettier {{path/to/file}}`
```

> - Check if the file has been formatted:

Reviewer 2: ```suggestion
- Check if a specific file has been formatted:
```

> `prettier {{file}} --check`

Reviewer 2: We use to put the argument to the left
```suggestion
`prettier --check {{path/to/file}}`
```

> - Give a configuration file:

Reviewer 2: ```suggestion
- Run with a specific configuration file:
```

> `prettier --config {{config}} {{file}}`

Reviewer 2: I think this could be used for a directory:
```suggestion
`prettier --config {{path/to/config_file}} {{path/to/file_or_directory}}`
```

- Format a file in place:

> `prettier {{file}} --write`

Reviewer 2: ```suggestion
`prettier --write {{path/to/file}}`
```

# prettier

> An opinionated code formatter for JavaScript, JSON, CSS, YAML, and more.
> More information: <https://prettier.io/>.

- Format a file and print the result to stdout:

> `prettier {{/path/to/file}}`

Reviewer 2: The `path/to/file` should be relative, not absolute:
```suggestion
`prettier {{path/to/file}}`
```

- Check if a specific file has been formatted:

`prettier --check {{path/to/file}}`

- Run with a specific configuration file:

`prettier --config {{path/to/config_file}} {{path/to/file}}`

> - Format a file in place:

Reviewer: I don't like how this is worded, and I can't tell what it does so I can't suggest anything

Reviewer 2: I don't it at all. @sbrl, could you help with this?

Reviewer 3: Hrm, let me see. It seems fine to me, but if others don't like it what about this?
```suggestion
- Format a file, replacing the original file with the formatted code:
```

Reviewer 4: `In place` could be a bit problematic, because I don't think many know that phrase. 
Another shorter suggestion:
```suggestion
- Format a file, replacing the original one:
```

Reviewer: what about

```suggestion
- Format a file, replacing the original:

`prettier --write {{path/to/file}}`

- Format all JavaScript files in the current directory recursively:

> `prettier --write '*.js'`

Reviewer: This is kind of pointless, and it isn't recursive through directories.

This command is just a longer version of the main command, and anyone should be able to identify the wildcard (*). 

It does have a use, but not on this page since it is common throughout *nix using *

Reviewer: (Just to make it easier for OP) 

```
```

Reviewer 2: Yeah, the command is not recursive, could be useful an example like `only format js files` because you're saying this format different files like YAML.

Could be useful an example with prettier settings in the CLI like:
```
prettier --single-quote --trailing-comma {{none}} --write {{path/to/directory}}
```

Reviewer: _If we decide to keep the command, it still needs to be changed since it implies that only JavaScript files can be done._

I could see the usefulness of that command suggested, however I would either remove it entirely or make it recursive by change it to */*/* or something similar ( I couldn't see a -R or -r flag)

Reviewer 3: To make it recursive you'd need to replace `'*.js'` with `"$(find . -type f -iname '*.js')"`, but that's dangerously close to demonstrating other commands' functionality. I recommend just updating the exampled description.

> - Use single quotes instead of double quotes and no trailing commas when formatting a file:

Reviewer 2: ```suggestion
- Format files or directories recursively using single quotes and no trailing commas:
```

> `prettier --single-quote --trailing-comma {{none}} --write`

Reviewer 2: Should we use `path/to/file_or_directory` here?

Reviewer 5: Same questiong.

Reviewer 3: Hrm. Why the `none` here?

Author: @sbrl the `none` is just for example, to show some command line formatting options.

Reviewer 2: @sbrl because we don't like trailing commas in JavaScript :rofl:

> `prettier --single-quote --trailing-comma {{none}} --write {{path/to/file_or_directory}}`

Reviewer: Is this recursive?

Reviewer 2: Yep https://prettier.io/docs/en/cli.html#file-patterns

> `prettier --single-quote --trailing-comma {{none}} --write {{path/to/file_or_directory`

Reviewer 2: ```suggestion
`prettier --single-quote --trailing-comma {{none}} --write {{path/to/file_or_directory}}`
```
